oeps/oep-0052/decisions/002-schema-specification.rst

```diff
@@ -0,0 +1,30 @@
+Event Bus Messages will be Defined in AVRO Specification Language
+=================================================================
+
+Context
+-------
+
+* Currently, Pulsar does not support `JSON Schema <https://json-schema.org/specification.html>`__, it only supports AVRO
+  for schema definition. Pulsar does support serializing AVRO to JSON (confusing using a ``JSONSchema`` class).
+
+* Industry best practices seem to suggest using a binary encoding of messages.(TODO, links)
+
+* Industry best practice also suggests using AVRO over JSONSchema for better expression around schema evolution. (TODO,
+  links)
+
+* Even if we use a JSON encoding for the message we'll probably want to enable compression on the wire to reduce data
+  size.  This would mean whether or not we use JSON encoding is moot because the data on the wire will still be binary
+  encoded.
+
+Decision
+--------
+
+* We will use the AVRO specification to define the schema of messages that are sent over the event bus.
```

Comment:
Is the plan to use the [Avro format for CloudEvents](https://github.com/cloudevents/spec/blob/v1.0.1/avro-format.md) conventions to be consistent with [OEP-41](https://open-edx-proposals.readthedocs.io/en/latest/oep-0041-arch-async-server-event-messaging.html), or is there a need to more drastically reassess OEP-41's recommendation of CloudEvents overall?

Reply:
I think this should be possible but we have not yet fully tried it yet.

Reply:
We also will need to assess the cost and benefit of trying to stick to CloudEvents, rather than just using whatever the tool provides out-of-the-box. For example, the [primitive Timestamp type for Pulsar](https://pulsar.apache.org/docs/en/schema-understand/#primitive-type) seems different from the CloudEvent spec. Will it be both possible and worth implementing around this? Unsure at this time.

Reply:
For now, we decided to stick to cloudevents convention as stated in OEP-41. You can see the specifics at https://github.com/eduNEXT/openedx-events/blob/main/openedx_events/avro_attrs_bridge.py#L196-L204

Reply:
@jinder1s: We should update the ADR with these details. Also see https://github.com/openedx/open-edx-proposals/pull/244#discussion_r781575760 (since I had forgotten about this thread).